app/models/service.rb

```diff
@@ -5,11 +5,12 @@ class Service < ActiveRecord::Base
   belongs_to :service_upload          # For bulk-uploaded services
   has_many :discontinued_services
 
-  validates_presence_of :recorded_by_educator_id, :student_id, :service_type_id, :recorded_at, :date_started#, :estimated_end_date
+  validates_presence_of :recorded_by_educator_id, :student_id, :service_type_id, :recorded_at, :date_started
+  validate :must_be_discontinued_after_service_start_date
 
   def discontinued?
-    (discontinued_services.size > 0) && !has_scheduled_end_date?  # If the end date is in the future
-                                                                  # the service isn't discontinued yet.
+    !discontinued_at.nil? && !has_scheduled_end_date? # If the end date is in the future
```

Comment:
Here @alexsoble please check my work for this model. I refactored a bit. I am just worried that I didn't get the original intent of the method right.

Reply:
Hey @Jngai! This is a method that will get a lot less complex now that we're flattening out the service model. 

I would say this method could look something like:

```
discontinued_at.present? && (Time.now > discontinued_at)
```

It's also good to try to avoid "double negative" expressions like `!discontinued_at.nil?` and replace them with "positive" expressions like `discontinued_at.present?`.